expression/builtin_arithmetic.go

```diff
@@ -61,17 +61,24 @@ const precIncrement = 4
 
 // numericContextResultType returns types.EvalType for numeric function's parameters.
 // the returned types.EvalType should be one of: types.ETInt, types.ETDecimal, types.ETReal
-func numericContextResultType(ft *types.FieldType) types.EvalType {
+func numericContextResultType(ctx sessionctx.Context, expr Expression) types.EvalType {
+	ft := expr.GetType()
+
 	if types.IsTypeTemporal(ft.Tp) {
 		if ft.Decimal > 0 {
 			return types.ETDecimal
 		}
 		return types.ETInt
 	}
+	evalTp4Ft := types.ETReal
 	if types.IsBinaryStr(ft) {
-		return types.ETInt
+		if expr.ConstItem(ctx.GetSessionVars().StmtCtx) {
+			evalTp4Ft = types.ETInt
+		}
+		// if expr is not const, expr is real
```

Comment:
if expr is not const, expr is real

Is there any document or code in MySQL that can verify this statement?